internal/server/data/migrations_test.go

```diff
@@ -728,6 +728,29 @@ DELETE FROM settings WHERE id=24567;
 				// schema changes are tested with schema comparison
 			},
 		},
+		{
+			label: testCaseLine("2022-10-05T18:00:00"),
+			setup: func(t *testing.T, db WriteTxn) {
+				_, err := db.Exec("INSERT INTO identities(id, name, deleted_at) VALUES(1111, 'hello@example.com', ?)", time.Now())
+				assert.NilError(t, err)
+				_, err = db.Exec("INSERT INTO provider_users(provider_id, identity_id, email) VALUES(9999, 1111, 'hello@example.com')")
+				assert.NilError(t, err)
+				_, err = db.Exec("INSERT INTO provider_users(provider_id, identity_id, email) VALUES(9999, 2222, 'hello@example.com')")
+				assert.NilError(t, err)
+			},
+			cleanup: func(t *testing.T, db WriteTxn) {
+				_, err := db.Exec("DELETE FROM identities")
+				assert.NilError(t, err)
+				_, err = db.Exec("DELETE FROM provider_users")
+				assert.NilError(t, err)
+			},
+			expected: func(t *testing.T, db WriteTxn) {
+				var remainingProviderUser uid.ID
+				err := db.QueryRow(`SELECT identity_id FROM provider_users`).Scan(&remainingProviderUser)
+				assert.NilError(t, err)
+				assert.Equal(t, remainingProviderUser, uid.ID(2222))
```

Comment:
minor: I think this test could pass even when the first record is not deleted, if the DB decided to return the records in a different order than we expect. I'm not sure if that's possible. 
Another option might be to `SELECT count(*) from provider_users where identity_id = 1111` which should give us a count of 0.